libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/CohortStatsHeatmap.tsx

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import {
+  defaultModelAssessmentContext,
+  ModelAssessmentContext,
+  HeatmapHighChart,
+  JointDataset,
+  generateMetrics,
+  ErrorCohort
+} from "@responsible-ai/core-ui";
+import { localization } from "@responsible-ai/localization";
+import { IDropdownOption } from "office-ui-fabric-react";
+import React from "react";
+
+import { generateCohortsStatsTable, wrapYAxisLabels } from "./StatsTableUtils";
+
+interface ICohortStatsHeatmapProps {
+  cohorts: ErrorCohort[];
+  selectableMetrics: IDropdownOption[];
+  selectedMetrics: string[];
+}
+
+class ICohortStatsHeatmapState {}
+
+export class CohortStatsHeatmap extends React.Component<
+  ICohortStatsHeatmapProps,
+  ICohortStatsHeatmapState
+> {
+  public static contextType = ModelAssessmentContext;
+  public context: React.ContextType<typeof ModelAssessmentContext> =
+    defaultModelAssessmentContext;
+
+  public render(): React.ReactNode {
+    const columns: string[] = [
+      localization.ModelAssessment.ModelOverview.countColumnHeader
+    ];
+    columns.push(
+      ...this.props.selectableMetrics
+        .filter((element) =>
+          this.props.selectedMetrics.includes(element.key.toString())
+        )
+        .map((element) => {
+          return element.text;
+        })
+    );
+
+    // generate table contents
+    const cohortLabeledStatistics = generateMetrics(
+      this.context.jointDataset,
+      this.props.cohorts.map((errorCohort) =>
+        errorCohort.cohort.unwrap(JointDataset.IndexLabel)
+      ),
+      this.context.modelMetadata.modelType
+    );
+
+    const items = generateCohortsStatsTable(
+      this.props.cohorts,
+      this.props.selectableMetrics,
+      cohortLabeledStatistics,
+      this.props.selectedMetrics
+    );
+
+    return (
+      <HeatmapHighChart
+        configOverride={{
+          chart: {
+            height: this.props.cohorts.length * 40 + 120,
+            type: "heatmap"
+          },
+          colorAxis: {
+            max: 1,
+            maxColor: "#1634F6",
+            min: 0,
+            minColor: "#FFFFFF"
+          },
+          legend: {
+            enabled: false
+          },
+          series: [
+            {
+              borderWidth: 1,
+              colorKey: "colorValue",
+              data: items,
+              dataLabels: {
+                enabled: true
+              },
+              name: "Metrics",
+              type: "heatmap"
+            }
+          ],
+          title: {
+            align: "left",
+            text: localization.ModelAssessment.ModelOverview
+              .dataCohortsHeatmapHeader
+          },
+          tooltip: {
+            formatter() {
+              // to avoid semantic error during build cast point to any
+              if (
+                this.point.y === undefined ||
+                (this.point as any).value === undefined ||

```

Comment:
nit: might this be slightly better?

```
const pointValue = (this.point as any).value
if (
                this.point.y === undefined || pointValue === undefined || pointValue === null
  ) {
...
```

Reply:
actually you already do this below on line 107, so now you can remove that line too
```
const value = (this.point as any).value;
```